Tools/RemoteSynchronization/Program.cs

```diff
@@ -0,0 +1,815 @@
+            var parse_opt = new Option<bool>(aliases: ["--parse-arguments-only"], description: "Only parse the arguments and then exit", getDefaultValue: () => DEFAULT_PARSE_ARGUMENTS_ONLY);
+
+            var root_cmd = new RootCommand(@"Remote Synchronization Tool
+
+This tool synchronizes two remote backends. The tool assumes that the intent is
+to have the destination match the source.
+
+If the destination has files that are not in the source, they will be deleted
+(or renamed if the retention option is set).
+
+If the destination has files that are in the source but are different in size or
+has a newer timestamp, they will be overwritten by the source files.
+
+If the force option is set, the destination will be overwritten by the source,
+regardless of the state of the files. It will also skip the initial comparison,
+and delete (or rename) all files in the destination.
+
+If the verify option is set, the files will be downloaded and compared after
+uploading to ensure that the files are correct. Files that already exist in the
+destination will be verified before being overwritten (if they seemingly match).
+            ");
+            root_cmd.AddArgument(src_arg);
+            root_cmd.AddArgument(dst_arg);
+            root_cmd.AddOption(dry_run_opt);
+            root_cmd.AddOption(src_opts);
+            root_cmd.AddOption(dst_opts);
+            root_cmd.AddOption(verify_contents_opt);
+            root_cmd.AddOption(verify_get_after_put_opt);
+            root_cmd.AddOption(retry_opt);
+            root_cmd.AddOption(force_opt);
+            root_cmd.AddOption(retention_opt);
+            root_cmd.AddOption(confirm_opt);
+            root_cmd.AddOption(global_opts);
+            root_cmd.AddOption(log_level_opt);
+            root_cmd.AddOption(log_file_opt);
+            root_cmd.AddOption(progress_opt);
+            root_cmd.AddOption(parse_opt);
+
+            root_cmd.SetHandler((InvocationContext ctx) =>
+            {
+                var parsed = ctx.ParseResult;
+
+                Dictionary<string, object?> options = parsed.CommandResult.Command.Options.ToDictionary(x => x.Name, x => parsed.GetValueForOption(x));
+
+                Run(parsed.GetValueForArgument(src_arg), parsed.GetValueForArgument(dst_arg), options).Wait();
+            });
+
+            return await root_cmd.InvokeAsync(args);
+        }
+
+        /// <summary>
+        /// The main logic of the tool.
+        /// </summary>
+        /// <param name="src">The connection string for the source backend.</param>
+        /// <param name="dst">The connection string for the destination backend.</param>
+        /// <param name="options">Various options for the tool</param>
+        /// <returns>The return code for the main entry; 0 on success.</returns>
+        private static async Task<int> Run(string src, string dst, Dictionary<string, object?> options)
```

Comment:
Is it for compatibility that you want the options dictionary?

Reply:
Yes, that was to pass the parsed options in easier. However, with the new way, this has become the config record.